src/uint8_array.c

```diff
@@ -36,6 +38,12 @@ rcutils_uint8_array_init(
   uint8_array->allocator = *allocator;
 
   if (buffer_capacity > 0lu) {
+    if (buffer_capacity >= SIZE_MAX) {
+      RCUTILS_SET_ERROR_MSG("requested capacity for uint8_array too large");
+      uint8_array->buffer_capacity = 0lu;
+      uint8_array->buffer_length = 0lu;
+      return RCUTILS_RET_INVALID_ARGUMENT;
```

Comment:
https://ci.ros2.org/job/ci_linux/26499/testReport/junit/rclcpp/TestSerializedMessage/failed_init_throws/, the upper layer expects this as bad alloc failure. to keep the consistency, i think we should change this into `RCUTILS_RET_BAD_ALLOC`.

Reply:
https://github.com/ros2/rcutils/pull/527/commits/f37cffa6131154b01e8338dcd7527f7238c764a3 should fix the failures.